lang/py/src/avro/schema.py

```diff
@@ -405,6 +419,102 @@ def __eq__(self, that):
     return self.props == that.props
 
 #
+# Decimal Type
+#
+class DecimalSchema(LogicalSchema):
+  def __init__(self, decimal_type, precision, scale):
+    if decimal_type != 'fixed' and decimal_type != 'bytes':
```

Comment:
Instead of checking the type here, it's better to check the `schema_type` parameter in each subclass and make sure it's the correct value. For example, the `schema_type` in the `FixedDecimalSchema` can only be `fixed`.